packages/aws-rfdk/docs/upgrade/upgrading-0.38.md

```diff
@@ -0,0 +1,254 @@
+# Upgrading to RFDK v0.38.x or Newer
+
+Starting in RFDK v0.38.0, [Deadline Secrets Management](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/secrets-management/deadline-secrets-management.html) will be enabled on the
+`Repository` and `RenderQueue` constructs by default when using Deadline 10.1.19 and later. This can cause some issues with existing farm configurations because Deadline Secrets Management requires:
+
+- The internal protocol on the Render Queue to be HTTPS
+- The external protocol on the Render Queue to be HTTPS
+
+Additionally, if you plan on upgrading to RFDK 0.38 or newer and Deadline 10.1.19 or newer, but do **not** want to enable Deadline Secrets Management, you may have to make changes to your RFDK application so that
+RFDK does not automatically configure Deadline Secrets Management on your farm.
+
+## Upgrading RFDK and Enabling Deadline Secrets Management
+
+---
+
+_This section describes how to upgrade your RFDK farm while also enabling Deadline Secrets Management. If you would **not** like to enable Deadline Secrets Management, please skip to the
+next section "Upgrading Farms Without Enabling Deadline Secrets Management"._
+
+---
+
+Your `RenderQueue` will use HTTPS for both internal and external protocols for RFDK 0.38.0 if **EITHER** of the following are true:
+
+1. You **HAVE NOT** specified the `trafficEncryption` (TypeScript) / `traffic_encryption` (Python) property
+1. You **HAVE** specified the `trafficEncryption` (TypeScript) / `traffic_encryption` (Python) property **AND BOTH of the following are true**:
+    1. **EITHER** of the following are true:
+        1. You **HAVE NOT** specified the `trafficEncryption.internalProtocol` (TypeScript) / `traffic_encryption.internal_protocol` (Python)
+        1. You **HAVE** specified the `trafficEncryption.internalProtocol` (TypeScript) / `traffic_encryption.internal_protocol` (Python) and it is set to `ApplicationProtocol.HTTPS`
+    1. **EITHER** of the following are true:
+        1. You **HAVE NOT** specified the `trafficEncryption.externalTLS` (TypeScript) / `traffic_encryption.external_tls` (Python)
+        1. You **HAVE NOT** specified `trafficEncryption.externalTLS.enabled` (TypeScript) / `traffic_encryption.external_tls.enabled` (Python)
+        1. You **HAVE** specified `trafficEncryption.externalTLS.enabled` (TypeScript) / `traffic_encryption.external_tls.enabled` (Python) and it is set to `true` (TypeScript) / `True` (Python)
+
+If you have determined that your `RenderQueue` will use HTTPS for both internal and external protocols in RFDK 0.38.0, then all you need to do to enable Deadline Secrets Management is to upgrade
+your RFDK application to use 0.38.0 and re-deploy.
```

Comment:
I know the logic is not yet implemented, but we plan to change this to reflect our plan to have secrets management only enabled when using Deadline version 10.1.19 or higher. What do you think about changing these docs here in advance?

```suggestion
your RFDK application to use RFDK `0.38.x` and Deadline `10.1.19.x` or higher and re-deploy.
```

Also, does it make sense to move this paragraph below the example snippets? What do you think?

Reply:
Sounds good, and yeah it flows better when it's below the examples. Updated